src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -486,35 +485,86 @@ Transactor::apply()
 NotTEC
 Transactor::checkSign(PreclaimContext const& ctx)
 {
+    // do not check signature of inner batch txn
+    if (ctx.tx.isFlag(tfInnerBatchTxn))
+        return tesSUCCESS;
+
+    auto const idAccount = ctx.tx.getAccountID(sfAccount);
+
     // If the pk is empty, then we must be multi-signing.
     if (ctx.tx.getSigningPubKey().empty())
-        return checkMultiSign(ctx);
-
-    return checkSingleSign(ctx);
-}
+    {
+        STArray const& txSigners(ctx.tx.getFieldArray(sfSigners));
+        return checkMultiSign(ctx.view, idAccount, txSigners, ctx.j);
+    }
 
-NotTEC
-Transactor::checkSingleSign(PreclaimContext const& ctx)
-{
-    // Check that the value in the signing key slot is a public key.
+    // Check Single Sign
     auto const pkSigner = ctx.tx.getSigningPubKey();
     if (!publicKeyType(makeSlice(pkSigner)))
     {
         JLOG(ctx.j.trace())
             << "checkSingleSign: signing public key type is unknown";
         return tefBAD_AUTH;  // FIXME: should be better error!
     }
-
-    // Look up the account.
     auto const idSigner = calcAccountID(PublicKey(makeSlice(pkSigner)));
-    auto const idAccount = ctx.tx.getAccountID(sfAccount);
     auto const sleAccount = ctx.view.read(keylet::account(idAccount));
     if (!sleAccount)
         return terNO_ACCOUNT;
 
+    return checkSingleSign(
+        idSigner, idAccount, sleAccount, ctx.view.rules(), ctx.j);
+}
+
+NotTEC
+Transactor::checkBatchSign(PreclaimContext const& ctx)
+{
+    NotTEC ret = tesSUCCESS;
+    STArray const& signers{ctx.tx.getFieldArray(sfBatchSigners)};
+    for (auto const& signer : signers)
+    {
+        auto const idAccount = signer.getAccountID(sfAccount);
+
+        Blob const& pkSigner = signer.getFieldVL(sfSigningPubKey);
+        if (pkSigner.empty())
+        {
+            STArray const& txSigners(signer.getFieldArray(sfSigners));
+            if (ret = checkMultiSign(ctx.view, idAccount, txSigners, ctx.j);
+                !isTesSuccess(ret))
+                return ret;
+        }
+        else
+        {
+            if (!publicKeyType(makeSlice(pkSigner)))
+                return tefBAD_AUTH;
```

Comment:
Add a test to cover this line

Reply:
Bump

Reply:
Bump - I think it should be fairly doable to add a test for this line

Reply:
Circling back to this

Reply:
Its not possible. checkBatchSign (Signature) is done in preflight, checkBatchSign (Ledger) is done in preclaim. So its actually already checked. Added an exclusion to code coverage.